tests/unit/amo/components/TestStaticPages.js

```diff
@@ -0,0 +1,35 @@
+import { shallow } from 'enzyme';
+import React from 'react';
+
+import { AboutBase } from 'amo/components/StaticPages/About';
+import { ReviewGuideBase } from 'amo/components/StaticPages/ReviewGuide';
+import { getFakeI18nInst } from 'tests/unit/helpers';
+
+
+describe('About', () => {
+  function render() {
+    return shallow(
```

Comment:
This should use `shallowUntilTarget` (see an example here: https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestSearchFilters.js#L31) so you can test the actual exported component and not have to render the base one.

For these tests it matters less as you're just adding translate but it's our usual way to test a component that is `compose`'d with HOCs.

Reply:
@tofumatt I've converted to use this pattern, and now I'm seeing:

```    console.error node_modules/fbjs/lib/warning.js:36
      Warning: Failed context type: The context `i18n` is marked as required in `Translate[AboutBase]`, but its value is `undefined`.
          in Translate[AboutBase]
```

This error seems to be coming up in a few other tests using the same pattern.

Presumably this happens because the i18n object isn't reaching the underlying component. It's completely possible I'm missing something - but since the shallowUntilTarget isn't actually doing what the HoC would do (e.g. providing the i18n object) I'm struggling to understand the rationale for using it over testing the component directly.

Reply:
I guess it matters more for consistency when you *would* be testing state in component as well–I don't want someone to look at these tests and think "Oh that's how I should test things".

In this particular case it's more about best practices and fewer surprises than needing to do it because yes: `i18n` isn't passed through with translate using `shallowUntilTarget`. Though really it should be, that seems like a bug.